midp/gfx.js

```diff
@@ -423,19 +426,24 @@ var currentlyFocusedTextEditor;
         });
     }
 
-    function withFont(font, c, str) {
+    function measureWidth(c, str) {
+        return c.measureText(str).width | 0;
+    }
+
+    function withFont(font, c) {
         if (c.font != font.css) {
           c.font = font.css;
         }
-        return c.measureText(str).width | 0;
     }
 
     function withTextAnchor(g, c, anchor, x, y, str, cb) {
+        withFont(g.klass.classInfo.getField("I.currentFont.Ljavax/microedition/lcdui/Font;").get(g), c);
```

Comment:
Move `withFont` into the block where `measureWith`  is called to reduce the times of calling `withFont`.